### PR TITLE
Fix infinite recursion in Storage.replace call

### PR DESCRIPTION
**Pull request by annakocheshkova**
* [ ] Has `CHANGELOG.md` been updated?
* [X] Are tests passing locally?
* [X] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [X] Did you check UI tests on the sample app? They are not executed on CI.
* [X] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

One picture is worth a thousand words...
image

### CODE REVIEW

File: AppCenterDataStorage/AppCenterDataStorageTests/MSDataStoreTests.m [hunk 1]
```diff
@@ -243,6 +243,63 @@ - (void)testCreateWhenDataModuleDisabled {
...
+  [MSDataStore createWithPartition:kMSPartitionTest
+                        documentId:kMSDocumentIdTest
+                          document:mockSerializableDocument
+                 completionHandler:^(MSDocumentWrapper *data) {
+                   completionHandlerCalled = YES;
+                   actualDocumentWrapper = data;
+                   [expectation fulfill];
+                 }];
+  id<MSSerializableDocument> replaceMockSerializableDocument = [MSFakeSerializableDocument new];
+  [MSDataStore replaceWithPartition:kMSPartitionTest
```
Comment (dhei) [inline AppCenterDataStorage/AppCenterDataStorageTests/MSDataStoreTests.m]: I don't see a test case covering the replace API which takes `writeOptions:` as parameter. Please consider adding a test for it.

Comment (annakocheshkova) [reply AppCenterDataStorage/AppCenterDataStorageTests/MSDataStoreTests.m]: It is called implecitly with nil options. But I can add a test case for writeOptions, too :)

Comment (yochaude) [reply AppCenterDataStorage/AppCenterDataStorageTests/MSDataStoreTests.m]: Let's not block the PR on that, the read/write stuff is being tested at the data proxy level already.

Thanks for catching that Anna!

Comment (jaelim-ms) [reply AppCenterDataStorage/AppCenterDataStorageTests/MSDataStoreTests.m]: We decided to unblock this.

**Pull request merged**